### DISCUSSION TITLE
It's way too hard to figure out how to use unstable_parseMultipartFormData to get a file contents?

### DISCUSSION THREAD

**Discussion by tmcw** [remix-run/remix#8278] [Q&A] [answered]
So, I'm trying to write an `action` that can receive an uploaded text file, read the text in the file, and parse that text. I've found the [documentation for that](https://remix.run/docs/en/main/utils/parse-multipart-form-data#unstable_parsemultipartformdata) and gotten the `File` object, but the docs say:

> Would be useful to understand [the Browser File API](https://developer.mozilla.org/en-US/docs/Web/API/File) to know how to use this API.

And I am underwhelmed by that level of explanation. I've used the Browser File API a ton and understand it fairly deeply. But this obviously isn't that. Like - I have this `File` object, so traditionally I'd use a `FileReader` to read it. But Remix doesn't polyfill `FileReader` or provide it as an export. So the contents of this file remain elusive to me, and the documentation doesn't really help me solve that puzzle: the MDN docs aren't useful if you don't have the tools described in the MDN docs.

The docs for the [memoryHandler](https://remix.run/docs/en/main/utils/unstable-create-memory-upload-handler) upload handler don't help either - again, they just link to MDN, which reminds us that `File` contents are opaque without `FileReader`.

So, digging through the code I figure out that the File polyfill is from https://github.com/remix-run/web-std-io - reading through that code you arrive at `File.stream()`. Which is a great API! And a fresh one… so fresh that it's not listed on the MDN page 😆

**Comment by akomm** [comment, marked as answer]
A `File` is also a `Blob`. It provides the following API:
![grafik](https://github.com/remix-run/remix/assets/2409779/6c4631a4-f3ba-4594-a578-d50a64188247)


You can work with it differently, whatever you want to do with it, even actually read the plain text.
![grafik](https://github.com/remix-run/remix/assets/2409779/d52cfdf9-ad9c-4221-89c0-9d0d6d091401)
